identity/aziot-identityd/src/lib.rs

```diff
@@ -362,7 +363,10 @@ impl Server {
 					}
 				};
 				device
-			}
+			},
+			settings::ProvisioningType::None => {
+				return Err(Error::DeviceNotFound);
```

Comment:
If this returns an error, the `main function` (or `reprovision_device` function) will fail fast, which is not what we want. We want the opposite effect for the IS process with the introduction of the `None` provisioning type, IIRC. The `provision_device` needs to be modified to return an enum (with possible values of `NoProvisioning` and `Provisioned`(IoTHubDevice) instead of returning a `IoTHubDevice` struct with `"none"` values .

Reply:
Fixed